cli/cmd/compose/compose.go

```diff
@@ -41,8 +41,7 @@ func (o *projectOptions) addProjectFlags(f *pflag.FlagSet) {
 	f.StringVarP(&o.ProjectName, "project-name", "p", "", "Project name")
 	f.StringArrayVarP(&o.ConfigPaths, "file", "f", []string{}, "Compose configuration files")
 	f.StringVar(&o.EnvFile, "env-file", "", "Specify an alternate environment file.")
-	f.StringVar(&o.WorkingDir, "workdir", "", "Specify an alternate working directory")
-	// TODO make --project-directory an alias
+	f.StringVar(&o.ProjectDir, "project-directory", "", "Specify an alternate working directory\n(default: the path of the Compose file)")
```

Comment:
unfortunately we can't just break existing usage on ACI/ECS like this I think, we'll need to keep workdir an alias of this flag (confusing with the exec one I agree, but... we can put project-directory in front more visible of course, just support running with the --workdir alias)

Reply:
I understand it's a breaking change, but that's actually a bug fix.
The problem is that in the case of `exec` and `run` we can use both at the same time. So in this case it's broken.